src/cache_cf.cc

```diff
@@ -4538,25 +4531,15 @@ static void free_sslproxy_cert_adapt(sslproxy_cert_adapt **cert_adapt)
 static void parse_sslproxy_cert_sign(sslproxy_cert_sign **cert_sign)
 {
     char *al;
-    sslproxy_cert_sign *cs = (sslproxy_cert_sign *) xcalloc(1, sizeof(sslproxy_cert_sign));
     if ((al = ConfigParser::NextToken()) == NULL) {
-        xfree(cs);
         self_destruct();
         return;
     }
 
-    if (strcmp(al, Ssl::CertSignAlgorithmStr[Ssl::algSignTrusted]) == 0)
-        cs->alg = Ssl::algSignTrusted;
-    else if (strcmp(al, Ssl::CertSignAlgorithmStr[Ssl::algSignUntrusted]) == 0)
-        cs->alg = Ssl::algSignUntrusted;
-    else if (strcmp(al, Ssl::CertSignAlgorithmStr[Ssl::algSignSelf]) == 0)
-        cs->alg = Ssl::algSignSelf;
-    else {
-        debugs(3, DBG_CRITICAL, "FATAL: sslproxy_cert_sign: unknown cert signing algorithm: " << al);
-        xfree(cs);
-        self_destruct();
-        return;
-    }
+    const auto algName = Security::certSignAlgorithmId(al); // throws on errors
+
```

Comment:
Please remove this empty line -- `algName` exists just to initialize/configure `cs`. It would not be needed at all if sslproxy_cert_sign were a "real" class!

Reply:
done